src/cluster_legacy.c

```diff
@@ -3605,8 +3606,10 @@ void clusterSendPing(clusterLink *link, int type) {
         clusterNode *this = dictGetVal(de);
 
         /* Don't include this node: the whole packet header is about us
-         * already, so we just gossip about other nodes. */
-        if (this == myself) continue;
+         * already, so we just gossip about other nodes.
+         * Also, don't include the receiver. Receiver will not update its state
+         * based on gossips about itself. */
+        if (this == myself || (link->node && this == link->node)) continue;
```

Comment:
What's the point of checking `link->node` (not null) before checking `this == link->node`?

```suggestion
        if (this == myself || this == link->node) continue;
```

Reply:
That was unnecessary. Used your suggestion. Thanks!